src/main/scala/org/apache/spark/sql/TiStrategy.scala

```diff
@@ -180,11 +180,11 @@ class TiStrategy(context: SQLContext) extends Strategy with Logging {
   // and then we don't return (don't planLater) and plan the remaining all at once
   private def doPlan(source: TiDBRelation, plan: LogicalPlan): Seq[SparkPlan] = {
 
-    val aliasMap = mutable.HashMap[Expression, Alias]()
+    val aliasMap = mutable.HashMap[(String,Expression), Alias]()
     val avgPushdownRewriteMap = mutable.HashMap[ExprId, List[AggregateExpression]]()
     val avgFinalRewriteMap = mutable.HashMap[ExprId, List[AggregateExpression]]()
 
-    def toAlias(expr: Expression) = aliasMap.getOrElseUpdate(expr, Alias(expr, expr.toString)())
+    def toAlias(expr: AggregateExpression) = aliasMap.getOrElseUpdate((expr.aggregateFunction.prettyName,expr.children.head), Alias(expr, expr.toString)())
```

Comment:
There's a function ```semanticEquals``` defined in [[org.apache.spark.sql.catalyst.expressions.Expression]], which can be used to determine whether two expressions are the same, and it has been broadly used in the Spark framework. I think the logic behind this function can help us with our expression deduplication.

Suppose we execute this query ```select count(c_custkey + 1),count(1 + c_custkey) from customer```, if we use ```expr.children.head``` to distinguish child expressions, we may not regard ```c_custkey + 1``` and ```1 + c_custkey``` as the same(but actually they are deterministic and can be canonicalized to a same expression), resulting in duplicated push downs. But if we switch to ```semanticEquals```'s method, the problem can be solved.

Below is my solution : )
```
    val aliasMap = mutable.HashMap[(Boolean, Expression), Alias]()

    def toAlias(expr: AggregateExpression) = {
      if (!expr.deterministic) {
        Alias(expr, expr.toString())()
      } else {
        aliasMap.getOrElseUpdate(
          (expr.deterministic, expr.canonicalized), Alias(expr, expr.toString)()
        )
      }
    }
```